state/state.go

```diff
@@ -1387,3 +1398,101 @@ func escape(value string) string {
 	intermediate = strings.Replace(intermediate, "}", "\\}", -1)
 	return strings.Replace(intermediate, ",", "\\,", -1)
 }
+
+//UnmarshalYAML will unmarshal the helmfile yaml section and fill the SubHelmfileSpec structure
+//this is required to keep allowing string scalar for defining helmfile (maybe)
+func (hf *SubHelmfileSpec) UnmarshalYAML(unmarshal func(interface{}) error) error {
+
+	var tmp interface{}
+	if err := unmarshal(&tmp); err != nil {
+		return err
+	}
+
+	switch i := tmp.(type) {
+	case string: // single path definition without sub items
+		hf.Path = i
+	case map[interface{}]interface{}: // helmfile path with sub section
+		for k, v := range i {
+			switch key := k.(type) {
+			case string:
+				//get the path
+				if key == "path" {
+					hf.Path = v.(string)
+				} else if key == "selectors" {
+					if err := extractSelectorContent(hf, v); err != nil {
+						return err
+					}
+				} else {
+					hf.Path = key
+					//get the selectors if something is specified
+					if v != nil { //we have a path, now compute the selector
+						if err := extractSelector(hf, v); err != nil {
+							return err
+						}
+					} //else it is a path with ending semi-colon without anything else below and it is fine
+				}
+			default:
+				return fmt.Errorf("Expecting a \"string\" scalar for the helmfile collection but got: %v", key)
+			}
+		}
+	}
+	//since we cannot make sur the "console" string can be red after the "path" we must check we don't have
+	//a SubHelmfileSpec with only selector and no path
+	if hf.Selectors != nil && hf.Path == "" {
+		return fmt.Errorf("found 'selectors' definition without path: %v", hf.Selectors)
+	}
+
+	return nil
+}
+
+//extractSelector this will extract the selectors: from the helmfile section
+//this has been developed to only expect selectors: under the helmfiles for now.
+func extractSelector(hf *SubHelmfileSpec, value interface{}) error {
+	switch value := value.(type) {
+	case map[interface{}]interface{}:
+		for k, v := range value {
+			switch key := k.(type) {
+			case string:
+				if key == "selectors" {
+					return extractSelectorContent(hf, v)
+				} else { //not 'selectors' so error
+					return fmt.Errorf("Expecting a \"selectors\" mapping but got: %v", key)
+				}
+			default: //we where expecting a selector but go something else
+				return fmt.Errorf("Expecting a \"selectors\" mapping for string [-%v] but got: %v of type %T", hf.Path, key, key)
+			}
+		}
+	default:
+		return fmt.Errorf("Expecting a \"selectors\" mapping for string [-%v] but got: %v of type %T", hf.Path, value, value)
+	}
+	return nil
+}
+
+func extractSelectorContent(hf *SubHelmfileSpec, value interface{}) error {
+	switch selectors := value.(type) {
+	case string: //check the string is `inherits` else error
+		if selectors == InheritsYamlValue {
+			hf.Inherits = true
```

Comment:
Can be a nit but I'd suggest using:

```
helmfiles:
- path: path/to/subhelmfile.yaml
  selectorsInherited: true
```

So that the implementation becomes a bit simpler without the overloading of `selectors`.

Reply:
yes, I'll do that